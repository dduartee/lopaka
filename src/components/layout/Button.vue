<script lang="ts" setup>
import {computed, defineProps} from 'vue';

const props = defineProps<{
    active?: boolean;
    secondary?: boolean;
    filled?: boolean;
    danger?: boolean;
    success?: boolean;
    isIcon?: boolean;
    variant?: string;
    title?: string;
    disabled?: boolean;
}>();

const classNames = computed(() => ({
    'btn-primary': !props.secondary || props.active,
    'btn-outline': !props.active && !props.filled,
    'btn-secondary btn-error': props.danger,
    'btn-success': props.success,
    'btn-square text-2xl align-top': props.isIcon,
    [`btn-${props.variant}`]: props.variant,
}));
</script>
<template>
    <div
        class="tooltip tooltip-bottom capitalize no-animation"
        :class="{'pointer-events-none': disabled}"
        :data-tip="title"
    >
        <button
            class="btn btn-sm font-sans uppercase leading-none"
            :class="classNames"
            :disabled="disabled"
        >
            <slot></slot>
        </button>
    </div>
</template>
<style lang="css" scoped>
.tooltip-bottom:before {
    z-index: 10;
}
</style>
