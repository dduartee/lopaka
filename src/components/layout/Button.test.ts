import {mount} from '@vue/test-utils';
import {describe, expect, it} from 'vitest';
import Button from '/src/components/layout/Button.vue';

describe('Button', () => {
    it('renders a button element', () => {
        const wrapper = mount(Button, {
            slots: {
                default: 'Test',
            },
        });
        expect(wrapper.find('.btn.btn-primary').exists()).toBe(true);
        expect(wrapper.element).toMatchSnapshot();
    });

    it('emits a click event when clicked', async () => {
        const wrapper = mount(Button);
        await wrapper.trigger('click');
        expect(wrapper.emitted('click')).toBeTruthy();
    });
});
