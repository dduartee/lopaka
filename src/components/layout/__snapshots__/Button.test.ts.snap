// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders a button element 1`] = `
<div
  class="tooltip tooltip-bottom capitalize no-animation"
  data-v-3947d5d5=""
>
  <button
    class="btn btn-sm font-sans uppercase leading-none btn-primary btn-outline"
    data-v-3947d5d5=""
  >
    
    Test
    
  </button>
</div>
`;
